File: backend/app/api/schemas/documents.py
"""Schemas for GET/DELETE /api/v1/documents (SDD Section 10.3, 10.4)."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.core.constants import DocumentStatus


class DocumentSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    display_name: str
    size_bytes: int
    page_count: int | None
    chunk_count: int
    status: DocumentStatus
    failure_reason: str | None
    uploaded_by: uuid.UUID
    created_at: datetime
    ready_at: datetime | None


class DocumentListResponse(BaseModel):
    items: list[DocumentSummary]
    total: int
    page: int
    page_size: int


class DocumentDeleteResponse(BaseModel):
    document_id: uuid.UUID
    chunks_removed: int

File: frontend/src/types/documents.ts
// Mirrors backend/app/api/schemas/documents.py

import type { DocumentStatus } from './enums'

export interface DocumentSummary {
  id: string
  display_name: string
  size_bytes: number
  page_count: number | null
  chunk_count: number
  status: DocumentStatus
  failure_reason: string | null
  uploaded_by: string
  created_at: string
  ready_at: string | null
}

export interface DocumentListResponse {
  items: DocumentSummary[]
  total: number
  page: number
  page_size: number
}

export interface DocumentDeleteResponse {
  document_id: string
  chunks_removed: number
}
